File: python/Dosen.py
#import data dari file Siak
from Siak import Siak
#Dosen 
class Dosen(Siak):
    
    #constructor
    def __init__(self, nip = 0, fakultas = "", pend_terakhir = "", keahlian = "", nama ="", nik = 0, jenisKelamin = "", asalUniv = "", emailEdu = ""):
        #untuk mengakses variabel dan metode pada parent
        super().__init__(asalUniv, emailEdu, nama, nik, jenisKelamin)
        self.nip = nip
        self.fakultas = fakultas
        self.pend_terakhir = pend_terakhir
        self.keahlian = keahlian
       
    #setter 
    def setNip(self, nip):
        self.nip = nip
    
    def setFakultas(self, fakultas):
        self.fakultas = fakultas
        
    def setPend_terakhir(self, pend_terakhir):
        self.pend_terakhir = pend_terakhir
        
    def setKeahlian(self, keahlian):
        self.keahlian = keahlian
        
    #getter 
    def getNip(self):
        return self.nip
    
    def getFakultas(self):
        return self.fakultas
    
    def getPend_terakhir(self):
        return self.pend_terakhir
    
    def getKeahlian(self):
        return self.keahlian

File: c++/Program/Dosen.cpp
#include"Header.hh"

Dosen::Dosen(): Siak(){
}

Dosen::Dosen(int nik, string nama, string jenis_kelamin, string asal_univ, string email_edu) : Siak(nik, nama, jenis_kelamin, asal_univ, email_edu){
}

Dosen::Dosen(int nik, string nama, string jenis_kelamin, string asal_univ, string email_edu, int nip, string fakultas, string pend_terakhir, string keahlian) : Siak(nik, nama, jenis_kelamin, asal_univ, email_edu) {
    this->nip = nip;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

void Dosen::setNip(int nip){
    this->nip = nip;
}
int Dosen::getNip(){
    return this->nip;
}

void Dosen::setFakultas(string fakultas){
    this->fakultas = fakultas;
}
string Dosen::getFakultas(){
    return this->fakultas;
}

void Dosen::setPendTerakhir(string pend_terakhir){
    this->pend_terakhir = pend_terakhir;
}
string Dosen::getPendTerakhir(){
    return this->pend_terakhir;
}

void Dosen::setKeahlian(string keahlian){
    this->keahlian = keahlian;
}
string Dosen::getKeahlian(){
    return this->keahlian;
}

Dosen::~Dosen(){

}